src/Utilities/_DataValidation.php

```diff
@@ -65,6 +65,29 @@ function validateValueName(string $valueName): void
     }
 }
 
+if (!function_exists('validateItems')) {
+    function validateItems(array $items): void
+    {
+        foreach ($items as $item) {
+            if (empty($item)) {
+                throw new InvalidArgumentError("Items must be a non-empty array");
+            }
+        }
+
+    }
+}
+
+if (!function_exists('validateFieldsKeys')) {
+    function validateFieldsKeys(array $items): void
+    {
+        foreach ($items as $field => $value) {
```

Comment:
Same as above-- please make sure `$items` isn't empty before trying to iterate over it.